Vector/ViewController/SettingsViewController.m

```diff
@@ -506,16 +592,54 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
             
             cell = surnameCell;
         }
-        else if (row == USER_SETTINGS_EMAIL_ADDRESS_INDEX)
+        else if (userSettingsEmailStartIndex <= row &&  row < userSettingsNewEmailIndex)
         {
             MXKTableViewCellWithLabelAndTextField *emailCell = [self getLabelAndTextFieldCell:tableView];
             
             emailCell.mxkLabel.text = NSLocalizedStringFromTable(@"settings_email_address", @"Vector", nil);
+            emailCell.mxkTextField.text = account.linkedEmails[row - userSettingsEmailStartIndex];
             emailCell.mxkTextField.userInteractionEnabled = NO;
             
             cell = emailCell;
         }
-        else if (row == USER_SETTINGS_CHANGE_PASSWORD_INDEX)
+        else if (row == userSettingsNewEmailIndex)
+        {
+            MXKTableViewCellWithLabelAndTextField *newEmailCell = [self getLabelAndTextFieldCell:tableView];
+
+            // Render the cell according to the `newEmailEditingEnabled` property
+            if (!_newEmailEditingEnabled)
+            {
+                newEmailCell.mxkLabel.text = NSLocalizedStringFromTable(@"settings_add_email_address", @"Vector", nil);
+                newEmailCell.mxkTextField.text = nil;
+                newEmailCell.mxkTextField.userInteractionEnabled = NO;
+                newEmailCell.accessoryView = [[UIImageView alloc] initWithImage:[UIImage imageNamed:@"plus_icon"]];
+            }
+            else
+            {
+                newEmailCell.mxkLabel.text = nil;
+                newEmailCell.mxkTextField.text = newEmailTextField.text;
+                newEmailTextField = newEmailCell.mxkTextField;
+                newEmailCell.mxkTextField.userInteractionEnabled = YES;
+                newEmailCell.mxkTextField.keyboardType = UIKeyboardTypeEmailAddress;
+                newEmailCell.mxkTextField.delegate = self;
+
+                [newEmailCell.mxkTextField removeTarget:self action:@selector(textFieldDidChange:) forControlEvents:UIControlEventEditingChanged];
+                [newEmailCell.mxkTextField addTarget:self action:@selector(textFieldDidChange:) forControlEvents:UIControlEventEditingChanged];
+
+                [newEmailCell.mxkTextField removeTarget:self action:@selector(textFieldDidEnd:) forControlEvents:UIControlEventEditingDidEnd];
+                [newEmailCell.mxkTextField addTarget:self action:@selector(textFieldDidEnd:) forControlEvents:UIControlEventEditingDidEnd];
+
```

Comment:
We should probably use [actionsForTarget: forControlEvent] instead of removing systematically the target before adding it. I know this mechanism is reproduced in other places :(.

Another solution should be to consider [tableView:didEndDisplayingCell:forRowAtIndexPath] in order to remove useless self reference in removed cells